### DISCUSSION TITLE
Any thoughts on single speaker?

### DISCUSSION THREAD

**Discussion by kradkfl** [lucasnewman/nanospeech#5] [General]
I'm wondering, given the fully learned alignment approach here, whether you (@lucasnewman) think that it's possible to train a single speaker model? I'm likely going to give it a go on LJSpeech, but thought I get your thoughts on it before embarking :-)

**Comment by kradkfl** [comment]
Another question: is there a specific formulation of `vocab.txt` that I need to adhere to? Is it just characters, or is BPE style?

edit: just saw the comments in `train_duration.py`; looks like I can use a simple character based tokenizer for english, but for a non-english dataset I should use some sort of BPE?

**Reply by lucasnewman** [reply]
> Another question: is there a specific formulation of `vocab.txt` that I need to adhere to? Is it just characters, or is BPE style?
> 
> edit: just saw the comments in `train_duration.py`; looks like I can use a simple character based tokenizer for english, but for a non-english dataset I should use some sort of BPE?

It's just characters from the exact vocabulary of the training dataset. BPE should work fine intuitively, but I haven't experimented with it extensively. The UTF-8 tokenizer would guarantee you can tokenize anything, though in my experience the exact vocabularies converge a bit better.

**Comment by lucasnewman** [comment]
> I'm wondering, given the fully learned alignment approach here, whether you (@lucasnewman) think that it's possible to train a single speaker model? I'm likely going to give it a go on LJSpeech, but thought I get your thoughts on it before embarking :-)

Yes, I've done exactly that. It works but it's definitely takes longer to get basic text alignment in place. There are a couple of options to work around this:

1) If you have access to a higher quality model, you could generate new samples in the same voice -- there will be some generational loss in the new samples, but it might not matter if you have a bunch of clean samples as well.

2) Augment with a larger dataset, even if it's multi-speaker, and at inference time just keep the voice fixed to your reference single speaker speech. Word error rate tends to fall with higher CFG rates, so in a sense you're training pronunciation on the larger dataset that will improve word-forming for your speaker, ideally without losing the characteristics of your reference speech. If you wanted to, you could even do this as an unsupervised pretraining step (they do this in the E2 TTS paper), but I personally dislike having to train multiple models, so my advice would be to just try to do it all supervised with a larger multi-speaker dataset mixed with LJSpeech.

**Reply by kradkfl** [reply]
Quick question: when you say "WER tends to fall with higher CFG rates," does this relationship hold w.r.t audio conditioning CFG _and_ text conditioning CFG, or one in particular?

From context I'm guessing text conditioning CFG, but I could be misunderstanding what about it allows for better pronunciation.

**Reply by lucasnewman** [reply]
I meant specifically audio (as CFG drops the text conditioning), which is why I was saying you could do it with an unsupervised pretraining step. This is from the E2 TTS paper where they do something similar:

> In a subset of our experiments, we initialized E2 TTS models
using a pre-trained model in an unsupervised manner. This pre-
training was conducted on an anonymized dataset, which consisted
of 200,000 hours of unlabeled data. The pre-training protocol, which
involved 800,000 mini-batch updates, followed the scheme outlined
in [34].
